<launch>

  <!-- Input arguments -->
  <arg name="device" value="1843" doc="TI mmWave sensor device type [1443, 1642]"/>
  <arg name="config" value="3d" doc="TI mmWave sensor device configuration [3d_best_range_res (not supported by 1642 EVM), 2d_best_range_res]"/>
  <arg name="max_allowed_elevation_angle_deg" default="90" doc="Maximum allowed elevation angle in degrees for detected object data [0 > value >= 90]}"/>
  <arg name="max_allowed_azimuth_angle_deg" default="90" doc="Maximum allowed azimuth angle in degrees for detected object data [0 > value >= 90]}"/>
  <include file="$(find detector)/launch/camera.launch">
    <param name="camera_serial" value="0"/>
  </include>
  <!-- mmWave_Manager node -->
  <node pkg="ti_mmwave_ros2" type="ti_mmwave_ros2" name="ti_mmwave" output="screen">
    <param name="command_port" value="/dev/ttyACM0"  />
    <param name="command_rate" value="115200"   />
    <param name="data_port" value="/dev/ttyACM1"  />
    <param name="data_rate" value="921600"   />
    <param name="max_allowed_elevation_angle_deg" value="$(arg max_allowed_elevation_angle_deg)"   />
    <param name="max_allowed_azimuth_angle_deg" value="$(arg max_allowed_azimuth_angle_deg)"   />
    <param name="frame_id" value="ti_mmwave"/>
  </node>
  
  <!-- mmWaveQuickConfig node (terminates after configuring mmWave sensor) -->
  <node pkg="ti_mmwave_ros2" type="mmWaveQuickConfig" name="mmWaveQuickConfig" args="$(find ti_mmwave_ros2)/cfg/xwr18xx_AOP_profile_allRange_inclclutter.cfg" output="screen" />

  <!-- Static transform from map to base_radar_link for visualization of stand-alone mmWave sensor using Rviz -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="radar_to_camera" args=" 0 0 0 0 0 0 1 ti_mmwave camera">
  </node>

   <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ti_mmwave_ros2)/launch/rviz/ti_mmwave_camera_overlay.rviz"/>
</launch>